app/javascript/components/editor/TestRunSummaryContainer.tsx

```diff
@@ -54,23 +54,17 @@ export const TestRunSummaryContainer = ({
   const handleTimeout = useCallback(() => {
     channel.current?.disconnect()
   }, [channel])
-  const handleCancelling = useCallback(() => {
-    clearTimeout(timer.current)
-
-    fetchJSON(cancelLink, {
-      method: 'POST',
-    }).then(() => {
-      setTestRun({ ...testRun, status: TestRunStatus.CANCELLED })
-    })
-  }, [cancelLink, testRun, setTestRun])
   const handleCancelled = useCallback(() => {
     clearTimeout(timer.current)
 
     channel.current?.disconnect()
-  }, [channel, timer])
+  }, [channel])
+
   const cancel = useCallback(() => {
     setTestRun({ ...testRun, status: TestRunStatus.CANCELLED })
-  }, [testRun, setTestRun])
+
```

Comment:
I made this an optimistic change. I mark the test run as cancelled even though we're not sure if the backend has acknowledged it yet.

Reply:
Yeah, that's fine.